WPF/SeeShells/SeeShells/ShellParser/ShellBagParser.cs

```diff
@@ -20,8 +21,19 @@ public ShellBagParser(IRegistryReader registryReader)
         /// <returns>a list of different variety ShellBag items</returns>
         public List<IShellItem> GetShellItems()
         {
-            // TODO: actual logic
-            return new List<IShellItem>();
+            List<IShellItem> shellItems = new List<IShellItem>();
+            foreach (RegistryKeyWrapper keyWrapper in registryReader.GetRegistryKeys())
+            {
+                if(keyWrapper.getValue() != null && keyWrapper.getValue().Length > 6) // Some Registry Keys are null
```

Comment:
@klaki892 I need your help with figuring out what exactly the issue is with the length of one of our raw binary arrays from the offline hive.
Take out the `&& keyWrapper.getValue().Length > 6` from the if statement and run the `GetShellItemsOffineTest` test.
My understanding is that the offset causes the unpacking method to start reading out of bounds but I can't quite put my finger on what exactly is going on.

Reply:
You're initial idea is correct, the offset is causing the upacking to read out of bounds
After debugging, the issue occurs when a byte array of `0x03 0x00 0x00 0x00` is passed to the ShellItemList. 
Its quite simple, as the code is written line 59 of `ShellItemList.cs` will add the size of the last item to the offset. We Simply dont check if the offset is equal to the size of the byte array before looking up the next size of the item. 
The fix is simply to put the following  lines ahead of line 47 in `ShellItemList`:
```
               //prevent out of bounds reading when the exact size of an item meets the next expected offset
                if (size == off && off != 0)
                    break;
```
The offline test case passed with this change and the one mentioned below.

Also dont do length < 6 there are actually valid shellitems that might only have 4 bytes (see ShellItem0x00) ShellItems themselves will tell you if they are empty (by size being 00)
```suggestion
                if(keyWrapper.getValue() != null ) // Some Registry Keys are null
```

Reply:
Fixed!